src/app/common/resolvers/dataset-view/dataset-view.resolver.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { inject } from "@angular/core";
+import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from "@angular/router";
+import { catchError, combineLatest, EMPTY, map, of, switchMap } from "rxjs";
+import { DatasetService } from "src/app/dataset-view/dataset.service";
+import ProjectLinks from "src/app/project-links";
+import { NavigationService } from "src/app/services/navigation.service";
+import { DatasetViewData, DatasetViewTypeEnum } from "src/app/dataset-view/dataset-view.interface";
+import { DatasetBasicsFragment } from "src/app/api/kamu.graphql.interface";
+import { activeTabResolver } from "../active-tab.resolver";
+import { DatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+
+export const datasetViewResolver: ResolveFn<DatasetViewData> = (
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot,
+) => {
+    const datasetService = inject(DatasetService);
+    const datasetSubService = inject(DatasetSubscriptionsService);
+    const navigationService = inject(NavigationService);
+    const accountName = route.paramMap.get(ProjectLinks.URL_PARAM_ACCOUNT_NAME) as string;
+    const datasetName = route.paramMap.get(ProjectLinks.URL_PARAM_DATASET_NAME) as string;
+    const activeTab = activeTabResolver(route, state);
```

Comment:
There is something wrong with this idea. `datasetViewResolver` is associated with the parent path, where we have ```/<account_name>/<dataset_name>``` prefix.  Logically it should not know anything about the "active tab". It should simply resolve basic dataset query.

Now, whether head has changed and we have to refresh, is not a responsibility of resolver..

Reply:
Fixed, removed extra code